static/sw.js

```diff
@@ -16,34 +14,53 @@ self.addEventListener("install", function (event) {
 self.addEventListener("fetch", function (event) {
   const requestUrl = new URL(event.request.url);
 
+  // Ignore non-HTTP/HTTPS requests
+  if (!requestUrl.protocol.startsWith("http")) {
+    return;
+  }
+
+  // Bypass cache for external image requests
+  if (
+    requestUrl.origin !== location.origin &&
+    /\.(jpe?g|png|webp)$/.test(requestUrl.pathname)
+  ) {
+    event.respondWith(fetch(event.request));
+    return;
+  }
+
   event.respondWith(
-    // Try to fetch from network first
-    fetch(event.request)
-      .then(function (networkResponse) {
-        // If request is successful, clone and cache the response
-        let cacheName = requestUrl.pathname;
-
-        // Choose the cache depending on the file type
-        if (/\.woff2?$|\.ttf$/.test(requestUrl.pathname)) {
-          cacheName = FONT_CACHE_CYF;
-        } else if (/\.jpe?g$|\.png$|\.webp$/.test(requestUrl.pathname)) {
-          cacheName = IMAGE_CACHE_CYF;
-        } else {
-          cacheName = CACHE_CYF;
-        }
-
-        caches.open(cacheName).then(function (cache) {
-          cache.put(event.request, networkResponse.clone());
-        });
+    caches.match(event.request).then(function (cachedResponse) {
+      var fetchPromise = fetch(event.request)
+        .then(function (networkResponse) {
+          if (!networkResponse || networkResponse.status !== 200) {
+            return cachedResponse;
+          }
+
+          let cacheName = CACHE_CYF;
+          let responseToCache = networkResponse.clone();
 
-        // Return the network response
-        return networkResponse;
-      })
-      .catch(function () {
-        // If network request fails, try to get from cache
-        return caches.match(event.request).then(function (cacheResponse) {
-          return cacheResponse;
+          // Choose the cache depending on the file type
+          if (/\.woff2?$|\.ttf$/.test(requestUrl.pathname)) {
+            cacheName = FONT_CACHE_CYF;
+          } else if (/\.jpe?g$|\.png$|\.webp$/.test(requestUrl.pathname)) {
+            cacheName = IMAGE_CACHE_CYF;
+          }
+
+          caches.open(cacheName).then(function (cache) {
+            cache.put(event.request, responseToCache);
+          });
+
+          return networkResponse;
+        })
+        .catch(function (error) {
+          console.log(
+            "[Service Worker] Network request Failed. Serving content from cache: " +
+              error
+          );
+          return cachedResponse;
         });
-      })
+
+      return fetchPromise || cachedResponse;
```

Comment:
It's been a long time since I studied the SW API, but wouldn't this give preference to the `fetchPromise` over the cache response?

If i understand correctly, that means we'd only serve the cache if the network fails - which seems quite similar (the same?) to what we were doing before?